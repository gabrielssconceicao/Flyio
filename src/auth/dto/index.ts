import { LoginDto } from './login.dto';
import { LoginResponseDto } from './login-response.dto';
import { RefreshTokenDto } from './refresh-token.dto';
import { TokenPayloadDto } from './token-payload.dto';

export { LoginDto, LoginResponseDto, RefreshTokenDto, TokenPayloadDto };
