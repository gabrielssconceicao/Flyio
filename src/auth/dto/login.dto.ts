import { IsNotEmpty, IsString } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';
import { ApiProperty } from '@nestjs/swagger';
export class LoginDto {
  @ApiProperty({
    example: 'emailOrUsername',
    description: 'Username or email',
  })
  @IsNotEmpty({
    message: dtoErrorMessages('Username or email', DtoErrorType.REQUIRED),
  })
  @IsString()
  login: string;

  @ApiProperty({ example: 'password', description: 'Password' })
  @IsNotEmpty({ message: dtoErrorMessages('Password', DtoErrorType.REQUIRED) })
  @IsString()
  password: string;
}
