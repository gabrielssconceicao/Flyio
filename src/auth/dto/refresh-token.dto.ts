import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString } from 'class-validator';

export class RefreshTokenDto {
  @ApiProperty({
    example: 'krcio48hgGYIFYJfuyfJYAYu',
    description: 'Refresh token',
  })
  @IsString()
  @IsNotEmpty()
  refreshToken: string;
}
