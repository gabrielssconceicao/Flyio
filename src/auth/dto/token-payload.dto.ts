export class TokenPayloadDto {
  sub: string;
  username: string;
  iat: number;
  exp: number;
  aud: string;
  iss: string;
}
