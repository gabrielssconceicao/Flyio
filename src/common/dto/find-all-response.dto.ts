import { ApiProperty } from '@nestjs/swagger';

export class FindAllResponseDto<T> {
  @ApiProperty({
    example: 1,
    description: 'Total count of items in the database',
  })
  count: number;

  @ApiProperty({
    description: 'List of items',
    isArray: true,
  })
  items: T[];
}
