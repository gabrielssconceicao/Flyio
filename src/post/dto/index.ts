import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
