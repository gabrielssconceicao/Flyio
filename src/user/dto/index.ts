import { CreateUserDto } from './create-user.dto';
import { FindAllUsersResponseDto } from './find-all-users.dto';
import { QueryParamDto } from './query-param.dto';
import { ReactivateUserDto } from './reactivate-user.dto';
import { UpdateUserDto } from './update-user.dto';

export {
  CreateUserDto,
  FindAllUsersResponseDto,
  QueryParamDto,
  ReactivateUserDto,
  UpdateUserDto,
};
