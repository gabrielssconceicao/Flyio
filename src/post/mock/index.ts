import { postMock } from './post.mock';
import { generateCreatePostDtoMock } from './create-post.dto.mock';
export { postMock, generateCreatePostDtoMock };
