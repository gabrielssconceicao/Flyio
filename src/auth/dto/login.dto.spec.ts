import { validate } from 'class-validator';
import { generateLoginDtoMock } from '../mocks/login.dto.mock';
import { LoginDto } from './login.dto';
import { formatErrors } from '../../common/utils/dto-format.errors';
describe('<LoginDto>', () => {
  let dto: LoginDto;
  beforeEach(() => {
    dto = new LoginDto();
    dto.usernameOrEmail = generateLoginDtoMock().usernameOrEmail;
    dto.password = generateLoginDtoMock().password;
  });

  it('should validate a DTO', async () => {
    const errors = await validate(dto);
    console.log({ errors });
    expect(errors.length).toBe(0);
  });

  describe('<Property: usernameOrEmail>', () => {
    it('should fail if is empty', async () => {
      dto.usernameOrEmail = '';
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('usernameOrEmail');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
  });
  describe('<Property: password>', () => {
    it('should fail if is empty', async () => {
      dto.password = '';
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
  });
});
