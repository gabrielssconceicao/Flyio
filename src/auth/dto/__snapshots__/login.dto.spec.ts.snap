// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginDto> <Property: login> should fail if is empty 1`] = `
[
  {
    "constraints": {
      "isNotEmpty": "Username or email is required",
    },
    "property": "login",
    "value": "",
  },
]
`;

exports[`<LoginDto> <Property: password> should fail if is empty 1`] = `
[
  {
    "constraints": {
      "isNotEmpty": "Password is required",
    },
    "property": "password",
    "value": "",
  },
]
`;
